scripts/scripts.js

```diff
@@ -212,9 +213,160 @@ function decorateFeatImg(getMetadata) {
+    }
+  });
+}
+
+function decorateMasonryBrick() {
+  const bricks = document.querySelectorAll('.masonry-layout .brick');
+  if (!bricks || bricks.length === 0) return;
+
+  bricks.forEach((brick) => {
+    const links = brick.querySelectorAll('a');
+    if (links && links.length === 1) {
+      const link = links[0];
+      brick.setAttribute('data-link', link.href);
+      brick.addEventListener('click', () => {
+        window.location.href = link.href;
+      });
+    }
+  });
+}
+
+function cloneBlockToSidebar(sidebar, blockClass) {
+  const targetBlock = document.querySelector(`.${blockClass}`);
+  if (!targetBlock) return false;
+
+  const clonedBlock = targetBlock.cloneNode(true);
+  targetBlock.classList.add('hide-if-sidebar-visible');
+
+  sidebar.append(clonedBlock);
+
+  return clonedBlock;
+}
+
+async function initSidebar() {
+  const sidebar = document.createElement('div');
+  sidebar.classList.add('blog-wrapper-sidebar');
+
+  const sidebarArticleMeta = cloneBlockToSidebar(sidebar, 'article-meta');
+  await initSharingLinkFunction(sidebarArticleMeta);
+  cloneBlockToSidebar(sidebar, 'banner');
+  cloneBlockToSidebar(sidebar, 'tags');
+
+  return sidebar;
+}
+
+function moveRecommendedArticleAsFinalSection(main) {
+  const recBlock = main.querySelector('.recommended-articles');
+  if (!recBlock) return;
+
+  const section = document.createElement('div');
+  section.classList.add('section');
+  section.append(recBlock);
+  main.append(section);
+}
+
 const { loadArea, setConfig, getMetadata } = await import(`${miloLibs}/utils/utils.js`);
 
+async function setUpSidebarLayoutForArticlePage() {
+  const main = document.querySelector('main');
+  const childContent = document.querySelector('.content');
+  const isArticlePage = getMetadata('content-type') === 'article';
+
+  if (!isArticlePage || !childContent) return;
+
+  // all divs that holds content
+  const sections = document.querySelectorAll('.section');
+  if (!sections) return;
+
+  // select first section to hold all the content
+  const mainContent = childContent.closest('.section');
+  mainContent.classList.add('blog-wrapper-content');
+
+  sections.forEach((section, index) => {
+    const isArticleBanner = section.querySelector('.marquee') && index === 0;
+    const isMainContent = mainContent === section;
+    if (!isArticleBanner && !isMainContent) {
+      mainContent.appendChild(section);
+    }
+  });
+
+  // Put blog wrapper (wraps sidebar + content) first
+  const blogWrapper = document.createElement('div');
+  blogWrapper.classList.add('blog-wrapper');
+  main.insertBefore(blogWrapper, mainContent);
+
+  // Put sidebar + content into blog wrapper
+  const sidebar = await initSidebar();
+  blogWrapper.append(sidebar, mainContent);
+
+  // Move recommended-articles as last separate section if any
+  moveRecommendedArticleAsFinalSection(main);
+}
+
+async function buildProgressBar() {
+  const { createTag } = await import(`${miloLibs}/utils/utils.js`);
```

Comment:
This is going to force the loading of utils.js before any other code runs in scripts.js since `buildProgressBar` is called first in `loadPage`.  utils.js normally loads after scripts.

I'd recommend copying a simplified version of `createTag` into here instead.

Reply:
Also is this really required to load this early in page load?  Seems like the progressbar could be late loaded as it isn't vital to page interaction

Reply:
Thanks Chris! Moved articles related functions into `articles.js` & moved the `buildProgressBar` into `decorateArticlePageUI` which is runned after `await loadArea();`

Let me know if this needs to be updated further.